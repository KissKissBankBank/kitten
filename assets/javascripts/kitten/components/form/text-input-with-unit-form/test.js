import React from 'react'
import renderer from 'react-test-renderer'
import { TextInputWithUnitForm } from './index'

describe('<TextInputWithUnitForm />', () => {
  let component

  describe('by default', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <TextInputWithUnitForm
            buttonLabel="To contribute"
            inputId="donation-amount"
            inputLabel="Enter your amount"
            inputPlaceholder="Your amount"
            inputUnit="€"
          />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with error prop', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <TextInputWithUnitForm
            inputIsOnError
            errorMessage="Error message…"
          />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with disabled prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<TextInputWithUnitForm formIsDisabled />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with align prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<TextInputWithUnitForm align="center" />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
