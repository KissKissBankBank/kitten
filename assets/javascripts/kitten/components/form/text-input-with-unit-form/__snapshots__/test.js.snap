// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputWithUnitForm /> by default matches with snapshot 1`] = `
<form
  className="sc-gzVnrw hovhA"
  onSubmit={[Function]}
>
  <div
    className="k-Grid k-Form-TextInputWithUnitForm__grid k-Form-TextInputWithUnitForm__grid--center k-u-margin-top-triple k-u-margin-bottom-triple"
  >
    <div
      className="k-Grid__col--12 k-Grid__col--7@xs k-Grid__col--5@m"
    >
      <div
        className="sc-bdVaJa eGDgJX"
      >
        <label
          className="k-Label k-Label--micro"
          htmlFor="donation-amount"
          onClick={[Function]}
        >
          Enter your amount
        </label>
      </div>
      <div
        className="sc-bdVaJa chkwVs"
      >
        <div
          className="sc-bxivhb jQUNJx k-Form-TextInputWithUnit"
        >
          <input
            autoComplete="off"
            className="sc-bwzfXH kBmQlm k-Form-TextInput k-Form-TextInputWithUnit__input k-Form-TextInput--andromeda"
            defaultValue=""
            disabled={false}
            id="donation-amount"
            name="text"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Your amount"
            type="number"
          />
          <span
            className="k-Form-TextInputWithUnit__unit"
          >
            €
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="k-Grid"
  >
    <div
      className="k-Grid__col--12 k-u-align-center"
    >
      <button
        className="sc-EHOje dXaVFU k-Form-TextInputWithUnitForm__submit k-Form-TextInputWithUnitForm__grid--default"
        disabled={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="submit"
      >
        To contribute
      </button>
    </div>
  </div>
</form>
`;

exports[`<TextInputWithUnitForm /> with align prop matches with snapshot 1`] = `
<form
  className="sc-gzVnrw hovhA"
  onSubmit={[Function]}
>
  <div
    className="k-Grid k-Form-TextInputWithUnitForm__grid k-Form-TextInputWithUnitForm__grid--center k-u-margin-top-triple k-u-margin-bottom-triple"
  >
    <div
      className="k-Grid__col--12 k-Grid__col--7@xs k-Grid__col--5@m"
    >
      <div
        className="sc-bdVaJa eGDgJX"
      >
        <label
          className="k-Label k-Label--micro"
          htmlFor=""
          onClick={[Function]}
        >
          
        </label>
      </div>
      <div
        className="sc-bdVaJa chkwVs"
      >
        <div
          className="sc-bxivhb jQUNJx k-Form-TextInputWithUnit"
        >
          <input
            autoComplete="off"
            className="sc-bwzfXH kBmQlm k-Form-TextInput k-Form-TextInputWithUnit__input k-Form-TextInput--andromeda"
            defaultValue=""
            disabled={false}
            id=""
            name="text"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="number"
          />
          <span
            className="k-Form-TextInputWithUnit__unit"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="k-Grid"
  >
    <div
      className="k-Grid__col--12 k-u-align-center"
    >
      <button
        className="sc-EHOje dXaVFU k-Form-TextInputWithUnitForm__submit k-Form-TextInputWithUnitForm__grid--default"
        disabled={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="submit"
      >
        
      </button>
    </div>
  </div>
</form>
`;

exports[`<TextInputWithUnitForm /> with disabled prop matches with snapshot 1`] = `
<form
  className="sc-gzVnrw hovhA"
  onSubmit={[Function]}
>
  <div
    className="k-Grid k-Form-TextInputWithUnitForm__grid k-Form-TextInputWithUnitForm__grid--center k-u-margin-top-triple k-u-margin-bottom-triple"
  >
    <div
      className="k-Grid__col--12 k-Grid__col--7@xs k-Grid__col--5@m"
    >
      <div
        className="sc-bdVaJa eGDgJX"
      >
        <label
          className="k-Label k-Label--micro"
          htmlFor=""
          onClick={[Function]}
        >
          
        </label>
      </div>
      <div
        className="sc-bdVaJa chkwVs"
      >
        <div
          className="sc-bxivhb jQUNJx k-Form-TextInputWithUnit"
        >
          <input
            autoComplete="off"
            className="sc-bwzfXH kBmQlm k-Form-TextInput k-Form-TextInputWithUnit__input k-Form-TextInput--andromeda k-Form-TextInput--disabled"
            defaultValue=""
            disabled={true}
            id=""
            name="text"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="number"
          />
          <span
            className="k-Form-TextInputWithUnit__unit k-Form-TextInputWithUnit__unit--disabled"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="k-Grid"
  >
    <div
      className="k-Grid__col--12 k-u-align-center"
    >
      <button
        className="sc-EHOje dXaVFU k-Form-TextInputWithUnitForm__submit k-Form-TextInputWithUnitForm__grid--default"
        disabled={true}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="submit"
      >
        
      </button>
    </div>
  </div>
</form>
`;

exports[`<TextInputWithUnitForm /> with error prop matches with snapshot 1`] = `
<form
  className="sc-gzVnrw hovhA"
  onSubmit={[Function]}
>
  <div
    className="k-Grid k-Form-TextInputWithUnitForm__grid k-Form-TextInputWithUnitForm__grid--center k-u-margin-top-triple k-u-margin-bottom-triple k-u-margin-bottom-single--important"
  >
    <div
      className="k-Grid__col--12 k-Grid__col--7@xs k-Grid__col--5@m"
    >
      <div
        className="sc-bdVaJa eGDgJX"
      >
        <label
          className="k-Label k-Label--micro"
          htmlFor=""
          onClick={[Function]}
        >
          
        </label>
      </div>
      <div
        className="sc-bdVaJa CoxgX"
      >
        <div
          className="sc-bxivhb jQUNJx k-Form-TextInputWithUnit"
        >
          <input
            autoComplete="off"
            className="sc-bwzfXH kBmQlm k-Form-TextInput k-Form-TextInputWithUnit__input k-Form-TextInput--andromeda k-Form-TextInput--error"
            defaultValue=""
            disabled={false}
            id=""
            name="text"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="number"
          />
          <span
            className="k-Form-TextInputWithUnit__unit k-Form-TextInputWithUnit__unit--error"
          />
        </div>
      </div>
      <div
        className="sc-bdVaJa ggQxmM"
      >
        <span
          className="sc-ifAKCX cFlEyZ k-u-color-error k-u-size-micro k-u-weight-regular"
        >
          Error message…
        </span>
      </div>
    </div>
  </div>
  <div
    className="k-Grid"
  >
    <div
      className="k-Grid__col--12 k-u-align-center"
    >
      <button
        className="sc-EHOje dXaVFU k-Form-TextInputWithUnitForm__submit k-Form-TextInputWithUnitForm__grid--default"
        disabled={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="submit"
      >
        
      </button>
    </div>
  </div>
</form>
`;
